service/dap/server.go

```diff
@@ -811,28 +831,34 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 
 	// Prepare the debug executable filename, build flags and build it
 	if mode == "debug" || mode == "test" {
-		output := args.Output
-		if output == "" {
-			output = defaultDebugBinary
-		}
-		output = cleanExeName(output)
-		debugbinary, err := filepath.Abs(output)
-		if err != nil {
-			s.sendInternalErrorResponse(request.Seq, err.Error())
-			return
+		debugbinary := args.Output
+		if debugbinary == "" {
+			debugbinary = s.tempDebugBinary()
 		}
 
-		buildFlags := args.BuildFlags
+		if !filepath.IsAbs(debugbinary) {
+			o, err := filepath.Abs(debugbinary)
```

Comment:
Won't this be a no-op if the path is already absolute? Why have the extra if?

Reply:
Abs is not a no-op, strictly speaking. It also calls filepath.Clean. But I guess that's harmless in most cases. Done.